### PR TITLE
Existing api key secret option

### PR DESCRIPTION
**Pull request by endriu0**
Before: No option for existing secret with infracostAPIKey

After: Possibility to use existing secret or external secret

Test: Deploy with this branch / see if DB gets populated

### CODE REVIEW

**Review by aliscott — APPROVED** [review]

File: charts/cloud-pricing-api/templates/secrets.yaml [hunk 1]
```diff
@@ -10,7 +10,9 @@ metadata:
     {{- include "cloud-pricing-api.labels" . | nindent 4 }}
 type: Opaque
 data:
+  {{-if not .Values.existingSecretAPIKey }}
```
Comment (aliscott) [inline charts/cloud-pricing-api/templates/secrets.yaml:13]: This is causing an issue with the Helm lint GitHub action step. I think it just needs a space like `{{ -if ...}`

**Review by aliscott — COMMENTED** [review]
Thanks @endriu0 🙏. This looks good. One small comment and then this is good to go.

File: charts/cloud-pricing-api/templates/secrets.yaml (hunk 1 above)
Comment (endriu0) [reply charts/cloud-pricing-api/templates/secrets.yaml:13]: Ok let me rewrite to drop the not

Comment (endriu0) [reply charts/cloud-pricing-api/templates/secrets.yaml]: Yep you're absolutely right. Typo :)

**Pull request merged**